### PR TITLE
Fixed exception messages for server variable parsing

### PR DESCRIPTION
**Pull request by mikaelm12**
Main thing here is that some of the exception messages for the server variable parsing was incorrect as pointed out in https://github.com/aspnet/BasicMiddleware/issues/143. I also did some refactoring to make the server variable parsing for IIS Url rewrite and Apache mod rewrite a little more similar.

Not urgent, just thought I'd open the pr.

@BrennanConroy @Tratcher @natemcmaster  @muratg

### CODE REVIEW

File: src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public static PatternSegment FindServerVariable(string serverVariable)
                 case "REQUEST_URI":
                     return new UrlSegment();
                 default:
-                    throw new FormatException("Unrecognized server variable");
+                    throw new FormatException(Resources.FormatError_InputParserUnrecognizedParameter(serverVariable, context.Index));
```
Comment (natemcmaster) [inline src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs]: The code has become inconsistent on strings vs resources. We should add an issue to clean this up. (You don't need to add to this PR.)

**Comment by mikaelm12** [comment]
@muratg Is this okay to merge to dev, or should we wait?

**Comment by muratg** [comment]
@mikaelm12 yeah, it should be OK to merge to dev.

File: src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs (hunk 1 above)
Comment (mikaelm12) [inline src/Microsoft.AspNetCore.Rewrite/Internal/IISUrlRewrite/ServerVariables.cs]: Okay, I logged https://github.com/aspnet/BasicMiddleware/issues/153

**Pull request merged**